source/common/upstream/health_checker_impl.cc

```diff
@@ -334,11 +334,18 @@ void HttpHealthCheckerImpl::HttpActiveHealthCheckSession::onGoAway(
 
 HttpHealthCheckerImpl::HttpActiveHealthCheckSession::HealthCheckResult
 HttpHealthCheckerImpl::HttpActiveHealthCheckSession::healthCheckResult() {
-  uint64_t response_code = Http::Utility::getResponseStatus(*response_headers_);
+  const uint64_t response_code = Http::Utility::getResponseStatus(*response_headers_);
   ENVOY_CONN_LOG(debug, "hc response={} health_flags={}", *client_, response_code,
                  HostUtility::healthFlagsToString(*host_));
 
   if (!parent_.http_status_checker_.inRange(response_code)) {
+    // If the HTTP response code would indicate immediate failure AND the immediate health check
```

Comment:
I think the HTTP code doesn't indicate immediate failure ever? Just failure?

Reply:
HTTP code failure does cause immediate failure, but not the type of immediate failure we are talking about here. I will update the comment to be more clear.